linear_algebra/src/rank_of_matrix.py

```diff
@@ -0,0 +1,83 @@
+"""This is a python program that calculates the rank of a matrix"""
+
+
+def rank_of_matrix(matrix: list[list]) -> int:
+    """
+    Finds the rank of a matrix.
+    Args:
+        matrix (list of lists): The matrix as a list of lists.
+    Returns:
+        int: The rank of the matrix.
```

Comment:
Please do not repeat the data types from the function signature. The problem with doing so is that the reader becomes confused if one changes and the other does not. Either describe what these variables are for or remove these comments.

```suggestion
        matrix: The matrix as a list of lists.
    Returns:
        The rank of the matrix.
```